File: app/src/main/java/com/app/avy/ui/adapter/ManageSpinnerAdapter.java
package com.app.avy.ui.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import androidx.appcompat.widget.AppCompatTextView;
import com.app.avy.R;
import com.app.avy.database.word.Word;

import java.util.ArrayList;
import java.util.List;

public class ManageSpinnerAdapter extends ArrayAdapter<Word> {
    private List<Word> data;

    public ManageSpinnerAdapter(Context context, ArrayList<Word> data) {
        super(context, 0, data);
        this.data = data;
    }

    @Override
    public View getDropDownView(int position, View convertView, ViewGroup parent) {
        return getView(position, convertView, parent);
    }

    @Override
    public View getView(final int position, View convertView, ViewGroup parent) {
        if (convertView == null) {
            convertView = LayoutInflater.from(getContext()).inflate(R.layout.item_manage_spinner, parent, false);
            convertView.setTag(ViewHolder.createViewHolder(convertView));
        }
        final ViewHolder holder = (ViewHolder) convertView.getTag();
        if (data.get(position).getSelect()) {
            holder.textCountry.setText(data.get(position).getMWord());
        }
        return convertView;
    }

    @Override
    public int getCount() {
        return data.size();
    }

    private static class ViewHolder {
        AppCompatTextView textCountry;

        static ViewHolder createViewHolder(View view) {
            ViewHolder holder = new ViewHolder();
            holder.textCountry = view.findViewById(R.id.tv_key);
            return holder;
        }
    }
}

File: app/src/main/java/com/app/avy/ui/adapter/ManageSpinnerAdapter.kt
package com.app.avy.ui.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import androidx.appcompat.widget.AppCompatImageView

class ManageSpinnerAdapter(context: Context, res: Int, list: ArrayList<String>?) :
    ArrayAdapter<String>(context, res, list) {

    var resource: Int
    var mData: ArrayList<String>? = null
    var vi: LayoutInflater

    init {
        this.resource = res
        this.mData = list
        this.vi = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var holder: ViewHolder
        var retView: View? = null
        if (convertView == null) {
            retView = vi.inflate(resource, parent, false)
            retView.tag = ViewHolder.createViewHolder(retView)
        }
        holder = ViewHolder()

        return retView!!
    }


    class ViewHolder {
        lateinit var img_chose: AppCompatImageView
        companion object {
            fun createViewHolder(view: View): ViewHolder {
                val holder = ViewHolder()
                return holder
            }
        }
    }
}